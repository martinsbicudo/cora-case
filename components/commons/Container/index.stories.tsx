import { StoryObj, Meta } from '@storybook/react'

import Container from '.'

export default {
  title: 'Commons / Container',
  component: Container,
} as Meta

export const Default: StoryObj = {
  args: {
    children: 'children container',
  },
}
