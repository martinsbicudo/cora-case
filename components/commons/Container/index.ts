import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  padding: 0 15px;
  max-width: 1024px;
  margin: 0 auto;
`

export default Container
