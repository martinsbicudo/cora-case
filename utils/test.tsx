import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
