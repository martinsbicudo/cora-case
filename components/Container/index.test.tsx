import { render, screen } from '@/utils/test'

import Container from '.'

describe('Components / Container', () => {
  const testId = 'children-container'
  const renderComponent = () =>
    render(
      <Container>
        <p data-testid={testId}>test</p>
      </Container>
    )

  it('should renders component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should find children', () => {
    renderComponent()
    const children = screen.getByTestId(testId)
    expect(children).toBeInTheDocument()
  })
})
