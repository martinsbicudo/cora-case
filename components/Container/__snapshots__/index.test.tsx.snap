// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Container should renders component 1`] = `
<div>
  <div
    class="sc-bcXHqe eNbcpT"
  >
    <p
      data-testid="children-container"
    >
      test
    </p>
  </div>
</div>
`;
